pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -472,7 +476,10 @@ class AnalyticsImpl implements Analytics {
   }
 
   @override
-  void close() => _gaClient.close();
+  Future<void> close() async {
```

Comment:
were we calling analytics.close() on the tool anyway?

I'm thinking (provided other clients aren't implementing this interface), an easier, non-breaking change to make would be to not change the signature of any existing methods, and just adding a new method, `waitForLastPing({Duration? timeout})` that clients can choose to opt in to, if they want to ensure an event was sent, or ignore if they'd rather prioritize exiting as soon as possible.

Reply:
> were we calling analytics.close() on the tool anyway?

No, and we didn't really have to because we weren't sending events yet, but that is a good catch for the [open PR](https://github.com/flutter/flutter/pull/136647) for me to add in the close statement on exit.

> non-breaking change to make would be to not change the signature of any existing methods

And i think this entire PR would be a breaking change regardless since the send methods return type has now changed

Reply:
I'm saying if we DON'T change the return signature of any existing methods, it won't be a breaking change.

Reply:
Oh I see, you don't agree on having the send method return `void` makes sense now though?  I understand it is a breaking change but I think this gets us more in line with the legacy analytics

Reply:
I'm not saying it doesn't make sense, I'm just pointing out the minimal amount of engineering effort would likely be to avoid breaking other clients. If think it's reasonable to force existing clients to migrate, then I'm ok with a breaking change.